pkg/combination/combination.go

```diff
@@ -96,8 +149,9 @@ func (cs *stream) All() ([]map[string]string, error) {
 		}
 		return nil, ErrCombinationsNotSolved
 	}
-
-	return cs.combinations, nil
+	combo := cs.combinations[0]
+	cs.combinations = cs.combinations[1:]
```

Comment:
Do we need to length check the `cs.combinations` variable in order to avoid attempting to blindly index into the array and producing a panic?

Reply:
Nope. The only time this would fail is `len(cs.combinations) == 0` which is checked just before this. If there is only one element originally `cs.combinations[1:]` will just be an empty array.